File: C_1_INT_TRANSFORM/main.cpp
#include <iostream>
#include <fstream>
#include <sstream>
#include <string>
#include <cstdlib>

int transform(int n, int p)
{
  // Convert into string
  std::string s = std::to_string(n);

  // Find the Pth digit
  int pthDigit = s[s.length() - p] - '0';

  // Build the transformed number as a string
  std::string transformed;
  for (int i = 0; i < s.length(); i++)
  {
    int digit = s[i] - '0';
    if (i < s.length() - p)
    {
      transformed += (digit + pthDigit) % 10 + '0';
    }
    else if (i > s.length() - p)
    {
      transformed += abs(digit - pthDigit) + '0';
    }
    else
    {
      transformed += s[i];
    }
  }

  return std::atoi(transformed.c_str());
}

int main()
{
  // Open the input file
  std::ifstream file("input.txt");
  if (!file.is_open())
  {
    std::cout << "Failed to open input.txt" << std::endl;
    return 1;
  }

  // Read the input from the file and test the transform function
  std::string line;
  while (std::getline(file, line))
  {
    std::istringstream iss(line);
    int n, p;
    iss >> n >> p;
    std::cout << transform(n, p) << std::endl;
  }

  return 0;
}

File: C_1_INT_TRANSFORM/main.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func transform(n int, p int) int {
	s := strconv.Itoa(n)

	pthDigit := int(s[len(s)-p] - '0')

	var transformed strings.Builder
	for i, c := range s {
		digit := int(c - '0')
		if i < len(s) - p {
			transformed.WriteByte(byte((digit + pthDigit) % 10 + '0'))
		} else if i > len(s) - p {
			transformed.WriteByte(byte(abs(digit - pthDigit) + '0'))
		} else {
			transformed.WriteByte(byte(digit + '0'))
		}
	}

	result, _ := strconv.Atoi(transformed.String())
	return result
}

func abs(x int) int {
	if x < 0 {
		return -x
	}
	return x
}

func main() {
	file, err := os.Open("input.txt")
	if err != nil {
		fmt.Println(err)
		return
	}
	defer file.Close()

	scanner := bufio.NewScanner(file)
	for scanner.Scan() {
		line := scanner.Text()
		parts := strings.Split(line, "  ")
		n, _ := strconv.Atoi(parts[0])
		p, _ := strconv.Atoi(parts[1])
		fmt.Println(transform(n, p))
	}
}